src/Tempest/Http/src/Routing/Construction/RouteMatchingRegexesBuilder.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Routing\Construction;
+
+use Tempest\Http\Routing\Matching\MatchingRegexes;
+
+final readonly class RouteMatchingRegexesBuilder
+{
+    // This limit is guesstimated using a small script with an ever in pattern feed into preg_match
+    private const int PREG_REGEX_SIZE_LIMIT = 32764;
```

Comment:
Wouldn't it make more sense if this was a factor of 2?

Reply:
I feel so stupid now. The limit IS a factor of 2. But I didn't specify that limit.

I found this value with an attached script. basically by increasing a pattern in `#^a$#` where the a is the size of the limit. To note that I have 4 extra characters in this regex. Thus my defined limit needs to be 4 bigger. which is a power of 2: `32768`.

Will fix.

Reply:
done